File: packages/common/erp-common-ms-utils-js/src/helpers/vault.ts
import vault from 'node-vault';
import { sendError, TypeErrorMessage } from '../errors';
import { axiosClient, AxiosInstance } from './axios';

export class Vault {
  private static _client: vault.client;
  static axios: AxiosInstance;

  static get client() {
    if (!Vault._client) {
      throw new Error('Can not get client. Please call Vault.init() first.');
    }
    return Vault._client;
  }

  static init = async (url: string, roleId: string, secretId: string) => {
    Vault.axios = axiosClient(url);
    const {
      data: {
        auth: { client_token },
      },
    } = await Vault.axios.post<{ auth: { client_token: string } }>('/v1/auth/approle/login', {
      role_id: roleId,
      secret_id: secretId,
    });
    Vault._client = vault({
      apiVersion: 'v1',
      endpoint: url,
      token: client_token,
    });
  };

  static read = async (path: string): Promise<{ data: { data: any } }> => {
    return await Vault.client.read(path);
  };
}

export const initVault = async () => {
  const vaultUrl = process.env.VAULT_URL!;
  const vaultRoleId = process.env.VAULT_ROLE_ID!;
  const vaultSecretId = process.env.VAULT_SECRET_ID!;
  if (!vaultUrl || !vaultRoleId || !vaultSecretId) {
    sendError(TypeErrorMessage.VAULT);
  }
  await Vault.init(vaultUrl, vaultRoleId, vaultSecretId);
};

File: packages/3/ms-3-flaskpy/flaskr/vault.py
import requests
import os
import hvac


class Vault:
    vault_client = None

    @staticmethod
    def init_vault(url, role_id, secret_id):
        data = requests.post(f"{url}/v1/auth/approle/login", json={'role_id': role_id, 'secret_id': secret_id})
        return data.json()

    @staticmethod
    def read(path):
        vault_secrets = Vault.vault_client.secrets.kv.v2.read_secret_version(mount_point='kv', path=path)['data']['data']
        return vault_secrets


def init_vault():
    vault_url = os.environ["VAULT_URL"]
    vault_role_id = os.environ["VAULT_ROLE_ID"]
    vault_secret_id = os.environ["VAULT_SECRET_ID"]

    data = Vault.init_vault(vault_url, vault_role_id, vault_secret_id)
    client_token = data['auth']['client_token']
    Vault.vault_client = hvac.Client(url=vault_url, token=client_token)

